sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -80,7 +80,18 @@ case class SortMergeJoinExec(
   override def requiredChildDistribution: Seq[Distribution] =
     ClusteredDistribution(leftKeys) :: ClusteredDistribution(rightKeys) :: Nil
 
-  override def outputOrdering: Seq[SortOrder] = requiredOrders(leftKeys)
+  override def outputOrdering: Seq[SortOrder] = joinType match {
+    case RightOuter =>
+      // For right outer join, values of the left key will be filled with nulls if it can't
+      // match the value of the right key, so `nullOrdering` of the left key can't be guaranteed.
+      // We should output right key order here.
+      requiredOrders(rightKeys)
+    case FullOuter =>
+      // Neither left key nor right key guarantees `nullOrdering` after full outer join.
+      Nil
+    case _ =>
```

Comment:
If possible, please use the white list. Otherwise, we might forget to update this when adding new join types. Then, we should issue the exception for the default case.